### DISCUSSION TITLE
What do you think of Avalonia XPF？

### DISCUSSION THREAD

**Discussion by xwang-888** [dotnet/wpf#7925] [Ideas]
What do you think of Avalonia XPF? Why don't officials work with Avalonia on cross-platform WPF, so that many cross-platform requirements will not consider QT.

**Comment by hahyes** [comment]
Avalonia XPF is a commercial product.

**Reply by xwang-888** [reply]
It is undeniable that the Avalonia team has implemented this function, while Microsoft, a giant company, has not developed in this direction since the birth of WPF, which has led many client developers to QT. I don't think Microsoft does not have the ability to develop cross-platform. I think MAUI's client development is very poor, at least compared with WPF. If WPF can be deployed on embedded devices, I think the framework will take on a new life.

**Reply by miloush** [reply]
See https://github.com/dotnet/wpf/issues/2967 and https://github.com/dotnet/wpf/issues/3952.

Ever heard of .NET Micro Framework? The UI was based on WPF.

**Reply by hahyes** [reply]
@xwang-888 I meant fact that this is a commercial product in early state, so there is not a lot of opinions about XPF. It is cool idea, but I wonder about compability in complex apps. Avalonia Team sadly shares no documentation about XPF except some marketing materials. Which is weird.

**Reply by lindexi** [reply]
@hahyes I think it is reasonable for a small development team to not have enough resources to write comprehensive documentation.

**Reply by MikeCodesDotNET** [reply]
I'm happy to set you up with a trial so you can test it yourself. 

So far, our customers are thrilled with XPF in terms of compatibility and performance. It's not a silver bullet, and some customers decide to migrate to Avalonia instead. 

The most challenging part (for us) is supporting all the 3rd party control vendors. While XPF is binary compatible with WPF, many control vendors use GDI and WinForms APIs, which we must support. You can check out our current 3rd party compatibility [here](https://avaloniaui.net/XPF/Packages), which should give you an idea of where we're at. We prioritise fixing controls based on the demand, so if a particular control blocks your trial, we'll fix it as quickly as possible.
